swim/heal_via_discover_provider.go

```diff
@@ -56,12 +58,12 @@ func newDiscoverProviderHealer(n *Node, baseProbability float64, period time.Dur
 		logger:           logging.Logger("healer").WithField("local", n.Address()),
 		started:          make(chan struct{}, 1),
 		quit:             make(chan struct{}),
+		rand:             rand.New(rand.NewSource(n.clock.Now().UnixNano())),
 	}
 }
 
 // Start the partition healing loop
 func (h *discoverProviderHealer) Start() {
-	rand.Seed(time.Now().UnixNano())
```

Comment:
If I understand correctly, removing this seed will cause the probability to kick in at the same time in the cluster.

In other words, probability to start partition healing will be exactly the same in the whole cluster, which increases the chances to start healing at the same time for all the nodes substantially.

Why do you need to remove the seed?